Docs/introducing-blazor-forms.md

```diff
@@ -0,0 +1,206 @@
+# Introducing Blazor Forms
+
+Forms are a key of any web application as they provide a way to collect information from end-users. This information might range from the user details that we collect as part of a sign-up process down to a single textbox that allows users to leave comments or post messages in an collaborative application.
+
+In this article we'll explain the benefits of Blazor forms, we'll look at their lifecycle and explain the fundamental concepts used to create and extend forms using Blazor.
+
+The article is divided into the following sections:
+
+1. [Form Fundamentals](#fundamentals)
+2. [Introducing Blazor Forms](#blazor-forms)
+3. [Blazor Component Model](#component-model)
+4. [Extending Form Components](#extending-components)
+5. [Validation](#validation)
+6. [Extending Validation](#extending-validation)
+
+## <a name="fundamentals"></a> Form Fundamentals
+
+Let's assume that we are building a Contact form which has two fields (Name and Email) and a button to submit the form. In ASP.NET MVC, the code required to build such a form might look like this.
+
+```html
+<form asp-controller="Home" asp-action="Register" method="post">
+  <label asp-for="Contact.Name" class="form-label">Name</label>
+  <input asp-for="Contact.Name" class="form-control" type="text" />
+  <span asp-validation-for="Contact.Name" class="text-danger"></span>
+
+  <label asp-for="Contact.Email" class="form-label">Email address</label>
+  <input asp-for="Contact.Email" class="form-control" />
+  <span asp-validation-for="Contact.Email" class="text-danger"></span>
+
+  <button class="btn btn-primary mt-3" type="submit">Submit</button>
+</form>
+```
+
+The form defines an HTTP method and an endpoint to which the data is sent. It uses ASP.NET tag helpers to render the correct types of input controls and validation messages based on data annotations on the underlying `Contact` model.
+
+One problem with this solution is that when the form submits, a full page refresh is required as the server now needs to send a full HTTP payload to the client. To avoid this, one could use the approach offered by SPA solutions which abstract and simplify common tasks such as managing the state of fragments of pages away from the developer.
+
+However, the separation of client and server-side code introduces further complexity, such as needing to deal with data type differences and other mismatches between server and client components.
+
```

Comment:
@dneimke I have refactored/simplified the above bit. Let me know what you think 😊

Reply:
Looks good, reads well!